spec/Plugin/AlwaysSeekableBodyPluginSpec.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace spec\Http\Client\Common\Plugin;
+
+use Http\Client\Common\Plugin;
+use Http\Client\Common\Plugin\AlwaysSeekableBodyPlugin;
+use Http\Client\Promise\HttpFulfilledPromise;
+use Http\Message\Stream\BufferedStream;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
+
+class AlwaysSeekableBodyPluginSpec extends ObjectBehavior
+{
+    public function it_is_initializable()
+    {
+        $this->shouldHaveType(AlwaysSeekableBodyPlugin::class);
+    }
+
+    public function it_is_a_plugin()
+    {
+        $this->shouldImplement(Plugin::class);
+    }
+
+    public function it_decorate_response_body_if_not_seekable(RequestInterface $request, ResponseInterface $response, StreamInterface $responseStream, StreamInterface $requestStream)
```

Comment:
we have issue #45 saying we should stop mocking request and response objects. i suggest to instantiate actual guzzlehttp/psr7 objects instead.

Reply:
i'm fine with #45 but here i need to test that some method on request / response are being (not) called, so i think it's better to keep the mocking here

Reply:
fair enough, then lets keep mocking.